mirrord/layer/src/exec.rs

```diff
@@ -112,6 +121,89 @@ fn intercept_tmp_dir(argv_arr: &Nul<*const c_char>) -> Detour<Vec<*const c_char>
     Bypass(TempDirEnvVarNotSet) // Temp dir was not set. Cannot intercept. Should not happen.
 }
 
+mod patched_args {
+    use std::ffi::CString;
+
+    use libc::c_char;
+
+    /// The arguments for `execve` or `posix_spawn` after patching the for side-stepping SIP.
+    /// Since the arguments for those functions are pointers, if we indeed modify the arguments, we
+    /// create new objects, so we have to keep them alive, so we put them in this struct, and the
+    /// detour code has to hold on to this struct until after the call to the original libc
+    /// functions.
+    pub struct PatchedSpawnArgs {
```

Comment:
I think `PatchedArgs` is okay since the context of this file is `exec`/`spawn`

Reply:
done.